app/src/controllers/plugins/sagas.js

```diff
@@ -252,6 +262,13 @@ function* watchPluginChange() {
   );
 }
 
+function* watchPublicPluginChange() {
+  yield takeEvery(
+    [createFetchPredicate(PUBLIC_PLUGINS), FETCH_GLOBAL_INTEGRATIONS_SUCCESS],
```

Comment:
`FETCH_GLOBAL_INTEGRATIONS_SUCCESS` is not necessary here as it is deprecated mechanism.
Use ` yield takeEvery(createFetchPredicate(PUBLIC_PLUGINS), fetchExtensionsMetadata)`
Inside of `fetchExtensionsMetadata` you can select necessary plugins based on action type which will be presented as parameter.